live-event-app/live-event-app/Theming/MessageListTheme.swift

```diff
@@ -0,0 +1,231 @@
+    dataDetectorTypes: .all,
+    linkTextAttributes: [:],
+    isEditable: false,
+    isExclusiveTouch: false,
+    scrollView: .disabled,
+    textContainerInset: .zero
+  ),
+  itemTheme: BasicComponentTheme(
+    imageView: ImageComponentTheme(customType: UIImageView.self, cornerRadius: 10),
+    primaryLabel: LabelComponentTheme(
+      customType: UILabel.self,
+      textFont: UIFont(name: "Poppins-Regular", size: 12),
+      textColor: UIColor(named: "MessageList.Cell.IncomingAuthorTextColor")!,
+      adjustsFontForContentSizeCategory: true,
+      textAlignment: .natural,
+      textMargin: .zero
+    ),
+    secondaryLabel: LabelComponentTheme(
+      customType: UILabel.self,
+      textFont: UIFont(name: "Poppins-Regular", size: 11),
+      textColor: UIColor(named: "MessageList.Cell.IncomingAuthorTextColor")!,
+      adjustsFontForContentSizeCategory: true,
+      textAlignment: .natural,
+      textMargin: .zero
+    )
+  ),
+  dateFormatter: .messageInline
+)
+
+let messageListComponentTheme = MessageListComponentTheme(
+  controllerType: CollectionViewComponent.self,
+  backgroundColor: .clear,
+  navigationBarTheme: navigationBarTheme,
+  collectionViewTheme:  messageListCollectionViewTheme,
+  messageInputComponent: messageInputComponentTheme,
+  incomingItemTheme: incomingItemTheme,
+  authorItemTheme: autorItemTheme,
+  typingIndicatorCellTheme: AnimatedTypingIndicatorCellTheme(
+    cellType: IMessageTypingIndicatorCell.self,
+    contentViewType: IMessageTypingBubbleView.self,
+    backgroundColor: .clear,
+    highlightColor: .clear,
+    selectedColor: .clear,
+    primaryContentColor: .clear,
+    secondaryContentColor: .clear,
+    animationEnabled: true,
+    pulsing: true,
+    bounces: true,
+    bounceDelay: 0.33,
+    bounceOffset: 0.25,
+    fades: true
+  ),
+  enableReactions: false
+)
+
+class CustomMessageListItemCell: MessageCollectionViewCellComponent {
+  
+  private var authorAvatarView: PubNubInlineAvatarComponentView!
+  private var messageContentView: PubNubMessageContentTextView!
+  
+  override func setupSubviews() {
+    super.setupSubviews()
+    
+    authorAvatarView = PubNubInlineAvatarComponentView()
+    messageContentView = PubNubMessageContentTextView()
+    
+    authorAvatarView.translatesAutoresizingMaskIntoConstraints = false
+    messageContentView.translatesAutoresizingMaskIntoConstraints = false
+    
+    cellContainer.axis = .horizontal
+    cellContainer.distribution = .fill
+    cellContainer.isLayoutMarginsRelativeArrangement = false
+    cellContainer.alignment = .center
+    
+    cellContainer.stackView.addArrangedSubview(authorAvatarView)
+    cellContainer.stackView.setCustomSpacing(10, after: authorAvatarView)
+    cellContainer.stackView.addArrangedSubview(messageContentView)
+    
+    authorAvatarView.widthAnchor.constraint(equalToConstant: 20).isActive = true
+    authorAvatarView.heightAnchor.constraint(equalToConstant: 20).isActive = true
```

Comment:
can we segregate this setup for author and message?

Reply:
This code is inside our `setupSubviews` method. How would you like to segregate this setup?